pubspec.yaml

```diff
@@ -110,4 +111,12 @@ flutter_native_splash:
 
   android_12:
     color: "#1B1B1B"
-    image: assets/android12splash.png
\ No newline at end of file
+    image: assets/android12splash.png
+
+
+flutter_icons:
+  android: "launcher_icon"
+  ios: true
+  image_path: "assets/android12splash.png"
+  adaptive_icon_background: "assets/icon_background.png"
+  adaptive_icon_foreground: "assets/android12splash.png"
```

Comment:
the icon should have it's own png file "assets/icon_launcher.png" instead of also using the splash icon to increase future modifiability

rerun of the script isn't necessary if the given path is correct since no changes to the icon should occur